### PR TITLE
br:tidy up the incompatible issues in using BR tools.

### PR DESCRIPTION
**Pull request by 3pointer**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Tidy up the incompatible issues in using BR tools.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by overvenus — COMMENTED** [review]
Could you inline `backup-and-restore-incompatible.md` into the section `## 兼容性`?

**Comment by 3pointer** [comment]
> Could you inline `backup-and-restore-incompatible.md` into the section `## 兼容性`?

Done, PTAL again

**Comment by 3pointer** [comment]
```
tiup/tiup-dm-topology-reference.md
	ERROR	https://github.com/pingcap/tiup/blob/master/examples/dm/topology.example.yaml
		Not Found (HTTP error 404)
```
it seems irrelevant 🤔️

**Review by overvenus — COMMENTED** [review]
LGTM

File: br/backup-and-restore-tool.md [hunk 1]
```diff
@@ -65,6 +65,35 @@ SST 文件以 `storeID_regionID_regionEpoch_keyHash_cf` 的格式命名。格式
...
+
++ 某些功能在开启或关闭状态下，会导致 KV 格式发生变化，因此备份和恢复期间没有统一开启或关闭，就会带来不兼容的问题
++ BR 部分版本和 TiDB 集群的接口不兼容
+
+下表整理了会导致 KV 格式发生变化的功能。
+
+功能 | 相关 issue | 解决方式
+|  ----  | ----  | ----- |
+聚簇索引 | [#565](https://github.com/pingcap/br/issues/565)       | 确保备份时 tidb_enable_clustered_index 变量和恢复时一致，否则会导致数据不一致的问题，如 default not found, 数据索引不一致。
+New collation  | [#352](https://github.com/pingcap/br/issues/352)       | 确保备份时集群 new_collations_enabled_on_first_bootstrap 和恢复时一致，否则会导致数据索引不一致，checksum 不通过。
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:80]: 备份是先于恢复吧？
```suggestion
New collation  | [#352](https://github.com/pingcap/br/issues/352)       | 确保恢复时集群的 new_collations_enabled_on_first_bootstrap 和备份时的一致，否则会导致数据索引不一致和checksum 不通过。
```

File: br/backup-and-restore-tool.md [hunk 2]
```diff
@@ -65,6 +65,35 @@ SST 文件以 `storeID_regionID_regionEpoch_keyHash_cf` 的格式命名。格式
...
+恢复集群开启 CDC 同步 | [#364](https://github.com/pingcap/br/issues/364#issuecomment-646813965) | BR ingest 的 SST 文件, TiKV 还没实现下推到 TiCDC，因此使用 BR 恢复时候需要关闭 TiCDC。
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:81]:
```suggestion
恢复集群开启 TiCDC 同步 | [#364](https://github.com/pingcap/br/issues/364#issuecomment-646813965) |  TiKV 暂不能将 BR ingest 的 SST 文件下推到 TiCDC，因此使用 BR 恢复时候需要关闭 TiCDC。
```

File: br/backup-and-restore-tool.md [hunk 3]
```diff
@@ -65,6 +65,35 @@ SST 文件以 `storeID_regionID_regionEpoch_keyHash_cf` 的格式命名。格式
...
+
+在上述功能确保备份恢复一致的**前提**下，BR 和 TiKV/TiDB/PD 还可能因为版本内部协议不一致/接口不一致出现不兼容的问题，因此 BR 内置了版本检查。
+
+### 版本检查
+
+BR 内置版本会在执行备份和恢复操作前，对 TiDB 集群版本和自身版本进行对比检查。如果大版本不匹配（比如 BR 4.x 和 TiDB 5.x 上），BR 会提示退出。如要跳过版本检查，可以通过设置 `--check-requirements=false` 强行跳过版本检查，但是可能会遇到版本不兼容的问题。
+需要注意的是，跳过检查可能会遇到版本不兼容的问题，现整理如下：
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:88]:
```suggestion
BR 内置版本会在执行备份和恢复操作前，对 TiDB 集群版本和自身版本进行对比检查。如果大版本不匹配（比如 BR 4.x 和 TiDB 5.x 上），BR 会提示退出。如要跳过版本检查，可以通过设置 `--check-requirements=false` 强行跳过版本检查，但是可能会遇到版本不兼容的问题。

需要注意的是，跳过检查可能会遇到版本不兼容的问题，现整理如下：
```

File: br/backup-and-restore-tool.md [hunk 4]
```diff
@@ -65,6 +65,35 @@ SST 文件以 `storeID_regionID_regionEpoch_keyHash_cf` 的格式命名。格式
...
+
+| 备份版本 \ 恢复版本| BR nightly / TiDB nightly| BR 5.0 / TiDB 5.0| BR 4.0 / TiDB 4.0 |
+|  ----  |  ----  | ---- | ---- |
+**BR nightly / TiDB nightly** | ✅ | ✅ | ✅ |
+**BR 5.0 / TiDB 5.0** | ✅ | ✅ | ✅
+**BR 4.0 / TiDB 4.0** | ✅ | ✅ | ✅(TiKV>=4.0.0-rc.1(BR 包含 [#233](https://github.com/pingcap/br/pull/233)，但 TiKV 不包含 [#7241](https://github.com/tikv/tikv/pull/7241)), 那么 BR 会导致 TiKV 节点重启) |
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:94]: **BR 包含 [#233](https://github.com/pingcap/br/pull/233)，但 TiKV 不包含 [#7241](https://github.com/tikv/tikv/pull/7241)**

是说“如果 BR 包含了 #233 的 Bug 修复但 TiKV 不包含 #7241 的 Bug 修复”吗？

Comment (3pointer) [reply br/backup-and-restore-tool.md:95]: 是说“如果 BR 包含了 #233 的 Bug 修复但 TiKV 不包含 #7241 的话"  会导致 TiKV 节点重启

File: br/backup-and-restore-tool.md [hunk 5]
```diff
@@ -65,6 +65,36 @@ SST 文件以 `storeID_regionID_regionEpoch_keyHash_cf` 的格式命名。格式
...
+
+在上述功能确保备份恢复一致的**前提**下，BR 和 TiKV/TiDB/PD 还可能因为版本内部协议不一致/接口不一致出现不兼容的问题，因此 BR 内置了版本检查。
+
+### 版本检查
+
+BR 内置版本会在执行备份和恢复操作前，对 TiDB 集群版本和自身版本进行对比检查。如果大版本不匹配（比如 BR 4.x 和 TiDB 5.x 上），BR 会提示退出。如要跳过版本检查，可以通过设置 `--check-requirements=false` 强行跳过版本检查，但是可能会遇到版本不兼容的问题。
+
+需要注意的是，跳过检查可能会遇到版本不兼容的问题，现整理如下：
+
+| 备份版本 \ 恢复版本| BR nightly / TiDB nightly| BR 5.0 / TiDB 5.0| BR 4.0 / TiDB 4.0 |
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:91]:
```suggestion
| 备份版本（纵向） \ 恢复版本（横向） | BR nightly / TiDB nightly| BR 5.0 / TiDB 5.0| BR 4.0 / TiDB 4.0 |
```

File: br/backup-and-restore-tool.md [hunk 6]
```diff
@@ -65,6 +65,36 @@ SST 文件以 `storeID_regionID_regionEpoch_keyHash_cf` 的格式命名。格式
...
+|  ----  |  ----  | ---- | ---- |
+**BR nightly / TiDB nightly** | ✅ | ✅ | ✅ |
+**BR 5.0 / TiDB 5.0** | ✅ | ✅ | ✅
+**BR 4.0 / TiDB 4.0** | ✅ | ✅ | ✅(TiKV>=4.0.0-rc.1(BR 包含 [#233](https://github.com/pingcap/br/pull/233)，但 TiKV 不包含 [#7241](https://github.com/tikv/tikv/pull/7241)), 那么 BR 会导致 TiKV 节点重启) |
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:95]:
```suggestion
| **BR nightly / TiDB nightly** | ✅ | ✅ | ✅ |
| **BR 5.0 / TiDB 5.0** | ✅ | ✅ | ✅
| **BR 4.0 / TiDB 4.0** | ✅ | ✅ | ✅（如果 TiKV>=4.0.0-rc.1，BR 包含 [#233](https://github.com/pingcap/br/pull/233) Bug 修复，且 TiKV 不包含 [#7241](https://github.com/tikv/tikv/pull/7241) Bug 修复，那么 BR 会导致 TiKV 节点重启) |
```

File: br/backup-and-restore-tool.md [hunk 7]
```diff
@@ -65,6 +65,36 @@ SST 文件以 `storeID_regionID_regionEpoch_keyHash_cf` 的格式命名。格式
...
+**BR nightly or 5.0 / TiDB 4.0** | ❌(TiDB < 4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌(TiDB < 4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌(TiDB < 4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) |
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:96]:
```suggestion
| **BR nightly 或 5.0 / TiDB 4.0** | ❌（当 TiDB < v4.0.9 时会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB < 4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB < 4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) |
```

**Review by TomShawn — COMMENTED** [review]
Global comment: BR 4.0 或 5.0 --> BR v4.0 或 v5.0，TiDB 4.0 或 5.0 相同

**Pull request merged**